.github/workflows/golangci-lint.yaml

```diff
@@ -0,0 +1,36 @@
+name: golangci-lint
+
+on:
+  pull_request:
+    branches: [ '*' ]
```

Comment:
don't we only want it to run when PR is created for `main` branch ?

Reply:
IMO we should run for all of them it will be useful at the time of backports.